From 7f6149449ca41ee520c0d5d222379b2081d97076 Mon Sep 17 00:00:00 2001
From: Glenn Washburn <development@efficientek.com>
Date: Mon, 10 Jul 2023 22:01:19 -0500
Subject: [PATCH 04/57] tests/util/grub-shell-luks-tester: Allow
 GRUB_SHELL_LUKS_DEFAULT_DEBUG and GRUB_TEST_DEFAULT_DEBUG to specify the
 debug level to grub-shell

Signed-off-by: Glenn Washburn <development@efficientek.com>
Reviewed-by: Daniel Kiper <daniel.kiper@oracle.com>
---
 tests/util/grub-shell-luks-tester.in | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tests/util/grub-shell-luks-tester.in b/tests/util/grub-shell-luks-tester.in
index 18920619a..850a20d6a 100644
--- a/tests/util/grub-shell-luks-tester.in
+++ b/tests/util/grub-shell-luks-tester.in
@@ -37,6 +37,8 @@ keyfile_offset=
 keyfile_size=
 KEYFILE_SIZE_MAX=4096
 
+debug="${GRUB_SHELL_LUKS_DEFAULT_DEBUG:-$GRUB_TEST_DEFAULT_DEBUG}"
+
 # Usage: usage
 # Print the usage.
 usage () {
@@ -350,7 +352,7 @@ if [ -n "$detached_header" ]; then
 fi
 
 # Run the test in grub-shell
-@builddir@/grub-shell ${debug:+--debug} $grub_shell_opts \
+@builddir@/grub-shell ${debug:+--debug=$debug} $grub_shell_opts \
     --modules="$modules" --qemu-opts="$qemuopts" \
     --files="${grub_testvars}=${testvars}" "$testcase" \
     >$testoutput
-- 
2.42.0

